src/components/FocusTrap/FocusTrapForModal/index.tsx

```diff
@@ -0,0 +1,9 @@
+import type FocusTrapForModalProps from './FocusTrapForModalProps';
+
+function FocusTrapForModal({children}: FocusTrapForModalProps) {
+    return children;
```

Comment:
What is the reason we sharded this based on platform? I see that focus-trap-react is only designed to work on web, but it is set up for SSR so it might be benign to keep it in place on native, which would simplify things a bit.

Furthermore, I'm wondering why the same focus-trapping behavior would be desired on web but not native. Maybe we should create a new package called `focus-trap-react-native` that uses RN primitives to achieve the same behavior, and then it would work on native (such as iPad w/ keyboard, for example). It's a pretty small package so that doesn't seem particularly high-effort.

Reply:
@roryabraham I don't think it's that easy to migrate `focus-trap-react` to react native. At first glance focus trap seems more like web concept as on mobile devices modals are rather presented as separate screens. Bottom sheet is a case that can be explored. 

There are also more things to consider. For example Android supports [keyboard navigation out of the box](https://developer.android.com/develop/ui/views/touch-and-input/keyboard-input/navigation) (although I don't think it works perfect in our App). Video below shows the example. Focus is not very visible. 

<details>
<summary>Keyboard navigation on Android</summary>


https://github.com/Expensify/App/assets/56261019/388ef278-ff0c-44f1-b648-8304f238ff0b


</details>

iOS on the other hand doesn't support keyboard navigation out of the box. It requires `Full Keyboard Access` accessibility setting to be enabled. It works quite well then but it also changes the UI with visible at all times blue focus. Please see the video  

<details>
<summary>Keyboard navigation on iOS</summary>


https://github.com/Expensify/App/assets/56261019/91d35564-b9cc-4b51-a4e4-8c10e1b75acb


</details>


In some places we manage arrow navigation by ourselves using `useArrowKeyFocusManager`. It has different behaviour then those shown above. 

<details>
<summary>Arrow manager focus on Android</summary>



https://github.com/Expensify/App/assets/56261019/d874d650-ed35-42ad-8ab7-f2d441edc73a




</details>


It seems that the topic is broader then just focus trap on mobile devices and it also includes keyboard navigation. So I think it shouldn't be explored as part of this task. What's your opinion @roryabraham?

Reply:
[Rory's response](https://swmansion.slack.com/archives/C04878MDF34/p1715183571751319?thread_ts=1715148894.780229&cid=C04878MDF34)